src/System.Management.Automation/utils/Encoding.cs

```diff
@@ -0,0 +1,399 @@
+/********************************************************************++
+Copyright (c) Microsoft Corporation.  All rights reserved.
+--********************************************************************/
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Management.Automation;
+using System.Management.Automation.Provider;
+
+namespace Microsoft.PowerShell
+{
+    /// <summary>
+    /// The list of available file encodings
+    /// </summary>
+    public enum FileEncoding
+    {
+        /// <summary>
+        /// No encoding, or unset.
+        /// </summary>
+        Unknown,
```

Comment:
Internally, we would have a const, similar to what the CLR does for colors (see [here](https://github.com/dotnet/corefx/blob/ef82e74c065a6f839515fee8871e1114e6aac357/src/System.Console/src/System/Console.cs#L170)).

I don't quite follow your question - but in C#, you can assign any value you like (with a cast). In PowerShell it's not as easy because we check an integer value to see if it's a valid enumerator.

We could also consider using the value `0` to mean `Unspecified` as long as the first enum starts at 1 - this has the benefit of being the default value if a field is not assigned.